### PR TITLE
FieldOnCorrectType validation rule

### PR DESCRIPTION
**Pull request by crisu83**

### CODE REVIEW

**Comment by Jalle19** [comment]
There are some namespace/autoloading issues that makes phpstan fail

**Comment by crisu83** [comment]
@Jalle19 yes, I just fixed them.

File: src/Validation/Rule/FieldOnCorrectTypeRule.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ * parent type, or are an allowed meta field such as __typename.
+ */
+class FieldOnCorrectTypeRule extends AbstractRule
+{
+    /**
+     * @inheritdoc
+     */
+    public function enterNode(NodeInterface $node): ?NodeInterface
+    {
+        if ($node instanceof FieldNode) {
```
Comment (Jalle19) [inline src/Validation/Rule/FieldOnCorrectTypeRule.php]: I would return early here to save one indent level

Comment (crisu83) [reply src/Validation/Rule/FieldOnCorrectTypeRule.php]: Well yes, @hungneox said the same. However, if you look at the rest of the rules I think it's easier to understand if we leave it like this.

**Pull request merged**